### PR TITLE
creating the new mock decorator and updating the config file

### PR DESCRIPTION
**Pull request by hrushikesh-s**
## Summary

Include a summary of major changes in bullet points:

- Feature 1: custom Mock decorator that mocks the class and its methods using unittest.mock
- Feature 2: all the device objects in alab_one that need to talk to the A-lab hardware (using API calls) have been mocked using a combination of get_driver() & connect() methods
- Feature 3: simulation tag added to CLI command in alab_management
- Fix 1: A-lab ecosystem can now be simulated just by programmatically chaining "alabos sim_mode --on" CLI command with "alabos launch" CLI command while launching the alab_management UI

## Checklist

Before a pull request can be merged, the following items must be checked:

- [X] Code is in the [standard Python style](https://www.python.org/dev/peps/pep-0008/).
  The easiest way to handle this is to run the following in the **correct sequence** on
  your local machine. Start with running [black](
  https://black.readthedocs.io/en/stable/index.html) on your new code. This will
  automatically reformat your code to PEP8 conventions and removes most issues. Then run
  [pycodestyle](https://pycodestyle.readthedocs.io/en/latest/), followed by [flake8](
  http://flake8.pycqa.org/en/latest/).
- [X] Docstrings have been added in the[Numpy docstring format](
  https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).
  Run [pydocstyle](http://www.pydocstyle.org/en/2.1.1/index.html) on your code.
- [X] Type annotations are **highly** encouraged. Run [mypy](http://mypy-lang.org/) to
  type check your code.
- [X] Tests have been added for any new functionality or bug fixes.
- [X] All linting and tests pass.

### CODE REVIEW

**Comment by hrushikesh-s** [comment]
This PR is ready for merging in master

File: alab_management/config.py [hunk 1]
```diff
@@ -83,7 +78,56 @@ def __init__(self):
             ) from exc
 
         self._path = Path(config_path).absolute()
-        self._config = freeze_config(_config)
+        # self._config = freeze_config(_config)
```
Comment (idocx) [inline alab_management/config.py:81]: #TODO: remove the private token

File: alab_management/sample_view/sample.py [hunk 2]
```diff
@@ -22,8 +22,8 @@ class Sample:
     task_id: Optional[ObjectId]
     name: str
     position: Optional[str]
-    metadata: Optional[Dict[str, Any]] = field(default_factory=dict)
-    tags: Optional[List[str]] = field(default_factory=list)
+    metadata: Optional[Dict[str, Any]] = field(default_factory=lambda: None)
```
Comment (idocx) [inline alab_management/sample_view/sample.py:25]: Why cannot it default to list/dict? That should be easier for searching in the MongoDB.

**Comment by hrushikesh-s** [comment]
@idocx @bernardusrendy, can you pls review this PR and merge if possible?

**Pull request merged**